File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-214358-38088/analysis/own-tests-project/src/main/java/surveycrawler/EndGuardedRunCommand.java
package surveycrawler;

public class EndGuardedRunCommand extends CrawlerCommand {

    // testing

    @Override
    public boolean isFor(char aCommand) {
        return aCommand == ')';
    }

    // executing

    @Override
    protected void executeOnce(SurveyCrawler aCrawler) {
        aCrawler.endGuardedRun();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-202806-59410/analysis/acceptance-project/src/surveycrawler/end_guarded_run_command.py
from surveycrawler.crawler_command import CrawlerCommand


class EndGuardedRunCommand(CrawlerCommand):

    # command character

    @classmethod
    def command_character(cls):
        return ')'

    # repeating

    def repeat(self, a_number_of_repetitions, a_crawler):
        a_crawler.signal_invalid_command()

    # executing

    def execute(self, a_crawler):
        a_crawler.end_guarded_run()
